op-guide/tidb-binlog.md

```diff
@@ -0,0 +1,323 @@
+---
+title: TiDB-Binlog 部署方案
+category: advanced
+---
+
+
+
+# TiDB-Binlog 部署方案
+
+## 目录
+
+  - [TiDB-Binlog 功能特性](#tidb-binlog-简介)
+  - [TiDB-Binlog 整体架构](#tidb-binlog-架构)
+  - [安装部署](#tidb-binlog-安装)
+  - [整体监控框架概述](https://github.com/pingcap/docs-cn/blob/master/op-guide/monitor-overview.md)
+  - [组件状态 API & 监控](#tidb-binlog-监控)
+
+## TiDB-Binlog 简介
+
+TiDB-Binlog 用于收集 TiDB 的 Binlog，并提供实时备份和同步功能的商业工具。
+
+TiDB-Binlog 支持以下功能场景:
+
+* *数据同步*:       同步 TiDB 集群数据到其他数据库
+* *实时备份和恢复*:  备份 TiDB 集群数据，同时可以用于 TiDB 集群故障时恢复
+
+## TiDB-Binlog 架构
+
+首先介绍 TiDB-Binlog 的整体架构。
+
+![TiDB-Binlog 架构](../media/architecture.jpeg)
+
+TiDB-Binlog 集群主要分为两个组件：
+
+### Pump 
+
+Pump 是一个守护进程，在每个 TiDB 的主机上后台运行。他的主要功能是实时记录 TiDB 产生的 Binlog 并顺序写入磁盘文件
+
+### Drainer
+
+Drainer 从各个 Pump 节点收集 Binlog，并按照在 TiDB 中事务的提交顺序转化为指定数据库兼容的 SQL 语句，最后同步到目的数据库或者写到顺序文件
+
+## TiDB-Binlog 安装
+
+### 下载官方 Binary
+
+- ( CentOS 7+ )
+
+```bash
+# 下载压缩包
+wget http://download.pingcap.org/tidb-binlog-latest-linux-amd64.tar.gz
+wget http://download.pingcap.org/tidb-binlog-latest-linux-amd64.sha256
+
+# 检查文件完整性，返回 ok 则正确
+sha256sum -c tidb-binlog-latest-linux-amd64.sha256
+
+# 解开压缩包
+tar -xzf tidb-binlog-latest-linux-amd64.tar.gz
+cd tidb-binlog-latest-linux-amd64
+```
+
+### TiDB-Binlog 部署
+
+- 推荐使用 ansible 部署 PUMP 
+
+*   搭建全新的 TiDB Cluster，启动顺序 pd-server -> tikv-server -> pump -> tidb-server -> drainer 
+    * 修改 tidb-ansible inventory.ini 文件
+        * enable_binlog = True
+    * 执行 ansible-playbook deploy.yml
+    * 执行 ansible-playbook start.yml
+        * drainer 目前需要手动部署
+
+*   对已有的 TiDB Cluster 部署 binlog 
+    * 修改 tidb-ansible inventory.ini 文件
+        * enable_binlog = True
+    * 执行 ansible-playbook rolling_update.yml
+        * drainer 目前需要手动部署
+
+### 注意
+
+*   需要为一个 TiDB 集群中的每台 TiDB server 部署一个 pump，目前 TiDB server 只支持以 unix socket 方式的输出 binlog。
+
+    我们设置 TiDB 启动参数 binlog-socket 为对应的 pump 的参数 socket 所指定的 unix socket 文件路径，最终部署结构如下图所示：
+
+    ![TiDB pump 模块部署结构](../media/tidb_pump_deployment.jpeg)
+
+*   drainer 不支持对 ignore schemas（在过滤列表中的 schemas） 的 table 进行 rename DDL 操作
+
+*   在已有的 TiDB 集群中启动 drainer，一般需要全量备份 并且获取 savepoint，然后导入全量备份，最后启动 drainer 从 savepoint 开始同步；
+
+    为了保证数据的完整性，在 pump 运行 10 分钟左右后按顺序进行下面的操作
+
+    *  以 gen-savepoint model 运行 drainer 生成 drainer savepint 文件，`bin/drainer -gen-savepoint --data-dir= ${drainer_savepoint_dir} --pd-urls=${pd_urls}`
+    *  全量备份，例如 mydumper 备份 tidb
+    *  全量导入备份到目标系统
+    *  设置 savepoint 文件路径，然后启动 drainer， `bin/drainer --config=conf/drainer.toml --data-dir=${drainer_savepoint_dir}`
+
+*   drainer 输出的 pb, 需要在配置文件设置下面的参数
+    ```
+    [syncer]
+    db-type = "pb"
+    disable-dispatch = true
+
+    [syncer.to]
+    dir = "/path/pb-dir"
+    ```
+
+
+### 示例及参数解释
+
+1.  Pump
```

Comment:
too many lines in a single ordered list item.

Is level 4 title suitable here?

``####``